### PR TITLE
Cleaning dockerfiles

### PR DESCRIPTION
**Pull request by ruffsl**
Take a look, ask questions,
Just wanted to get this to you quickly.

### CODE REVIEW

File: indigo/release/Dockerfile [hunk 1]
```diff
@@ -5,7 +5,6 @@ FROM osrf/ros:indigo-desktop
 MAINTAINER Dave Coleman dave@dav.ee
 
 # apt-commands are combined in single RUN statement with lists folder removal to reduce image size
-RUN apt-get update \
- && apt-get upgrade -y \
```
Comment (davetcoleman) [inline indigo/release/Dockerfile:9]: why is upgrade a bad idea?

File: indigo/release/Dockerfile [hunk 2]
```diff
@@ -5,7 +5,6 @@ FROM osrf/ros:indigo-desktop
...
- && apt-get install -y ros-indigo-moveit-full \
- && rm -rf /var/lib/apt/lists/*
\ No newline at end of file
+RUN apt-get update && apt-get install -y \
+      ros-${ROS_DISTRO}-moveit-full \
```
Comment (davetcoleman) [inline indigo/release/Dockerfile:9]: So I assume ${ROS_DISTRO} is set from the `FROM osrf/ros:indigo-desktop` image? cool

**Comment by davetcoleman** [comment]
This is awesome, thanks. I asked a few questions inline, out of curiosity.

File: indigo/release/Dockerfile (hunk 2 above)
Comment (ruffsl) [inline indigo/release/Dockerfile:9]: It's actually set in the official ros core image from the get go: `ENV ROS_DISTRO indigo`

https://github.com/osrf/docker_images/blob/8aed5cc7dc1bcd16b77e5014acc0fe481b6f4468/ros/indigo/indigo-ros-core/Dockerfile#L29

File: indigo/release/Dockerfile (hunk 1 above)
Comment (ruffsl) [inline indigo/release/Dockerfile:9]: It's customary to keep/use the packages from the parent image. If you upgrade, it defeats a lot of the purpose in the space saving benefits of building from a layered image, as now the resulting image would include a diff history of changed package files, which could get huge fast. If you do need to upgraded, or something breaks downstream, then generally it's about time to trigger any parent imges upstream to rebuild. 

For the **non release images**, I let that slide and left your update command. I figured getting the most recent ros environment for development purposes may be more important that being agile with images. **Although** if temporal reproducibility important to you in even the development source images, I'd encourage you omit the use of upgrade there as well. But this is a tradeoff you should think about or experiment with.

File: indigo/release/Dockerfile (hunk 2 above)
Comment (davetcoleman) [inline indigo/release/Dockerfile:9]: thanks for the explanation!

**Comment by ruffsl** [comment]
Let me know when you get around creating automated docker hub repos for these Dockerfiles when they find their final resting repo, I could help you set them up, plus auto trigger and what not for the Movit's organisations Docker Hub profile.

**Comment by davetcoleman** [comment]
I already have them [automated on dockerhub](https://hub.docker.com/r/davetcoleman/moveit_docker/builds/) - is that what you mean? I'm thinking of moving it to `ros-planning/moveit_docker` though I've been wondering if it would be better to be inline with the source code of the main moveit repo so that the triggers work correctly. Thanks for all your help!

File: indigo/release/Dockerfile (hunk 1 above)
Comment (davetcoleman) [inline indigo/release/Dockerfile:9]: i decided to remove it - i like the agility:
https://github.com/davetcoleman/moveit_docker/commit/bb2068edf4a095a27befb1f33f804d5fc9045b3f

**Comment by ruffsl** [comment]
Hmm, I think you're right in that it would be appropriate to keep the MoveIt! Dockerfiles under a `docker` subdirectory within the main MoveIt! repo. It's a common practice among many projects [e.g.](https://github.com/BVLC/caffe/tree/master/docker). 

I strongly recommend however you make a legitimate [DockerHub organization](https://hub.docker.com/organizations/add/) for MoveIt! and host the DockerHub repo under that, rather than your own personal account. This will make them more reputable, noticeable, centralized, as well as permit better management and collaboration, as you could delegate multiple users just as you can with GitHub Orgs.

**Comment by davetcoleman** [comment]
Yep, the personal account was temporary. Now: https://hub.docker.com/r/moveit/moveit_docker/

**Pull request merged**